File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        int has0 = -1;
        for (int i=0; i<n; i++) {
            int val = abs(nums[i]);
            if (val == 0) has0 = i;
            else nums[val-1] *= -1;
        }
        if (has0 == -1) return 0;
        for (int i=0; i<n; i++) {
            if (nums[i] > 0) return i+1;
        }
        return has0+1;
    }
};

File: 0268-missing-number/0268-missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        has0 = False
        n = len(nums)
        for i in range(n):
            j = abs(nums[i]) - 1
            if j < 0:
                has0 = True
            else:
                nums[j] *= -1
        if not has0:
            return 0
        idx0 = -1
        for i in range(n):
            if nums[i] > 0:
                return i + 1
            elif nums[i] == 0:
                idx0 = i + 1
        return idx0